pkg/ingester/wal.go

```diff
@@ -252,9 +269,10 @@ func (w *walWrapper) performCheckpoint() (err error) {
 		return errors.Wrap(err, "rename checkpoint directory")
 	}
 
-	// The last segment might still have been active during the checkpointing,
-	// hence delete only the segments before that.
-	if err := w.wal.Truncate(lastSegment - 1); err != nil {
+	// We delete the WAL segments which are before the previous checkpoint and not before the
+	// current checkpoint. This is because if the latest checkpoint is corrupted for any reason, we
+	// should be able to recover from the older checkpoint which would need the older WAL segments.
+	if err := w.wal.Truncate(lastCh - 1); err != nil {
```

Comment:
Here we do keep until `lastCh - 1` but then below we do `w.deleteCheckpoints(lastCh)` which deletes also the checkpoint `lastCh - 1`. Is this correct?

Reply:
After some offline discussion, this should be `lastCh` here

Reply:
`w.wal.Truncate(lastCh)` is the right thing here